pennylane_qulacs/qulacs_device.py

```diff
@@ -53,167 +86,186 @@ def hermitian(*args):
-                    'GPU not supported with installed version of qulacs. '
-                    'Please install "qulacs-gpu" to use GPU simulation.'
+                    "GPU not supported with installed version of qulacs. "
+                    "Please install 'qulacs-gpu' to use GPU simulation."
                 )
 
-            self._state = QuantumStateGpu(wires)
+            self._state = QuantumStateGpu(self.num_wires)
         else:
-            self._state = QuantumState(wires)
-
-        self._circuit = QuantumCircuit(wires)
-        self._first_operation = True
-
-    def apply(self, operation, wires, par):
-        if operation == 'BasisState' and not self._first_operation:
-            raise DeviceError(
-                'Operation {} cannot be used after other Operations have already been applied '
-                'on a {} device.'.format(operation, self.short_name)
-            )
-
-        self._first_operation = False
-
-        if operation == 'QubitStateVector':
-            if len(par[0]) != 2**len(wires):
-                raise ValueError('State vector must be of length 2**wires.')
-
-            self._state.load(par[0])
-        elif operation == 'BasisState':
-            if len(par[0]) != len(wires):
-                raise ValueError('Basis state must prepare all qubits.')
-
-            basis_state = 0
-            for bit in reversed(par[0]):
-                basis_state = (basis_state << 1) | bit
-
-            self._state.set_computational_basis(basis_state)
-        elif operation == 'QubitUnitary':
-            if len(par[0]) != 2 ** len(wires):
-                raise ValueError('Unitary matrix must be of shape (2**wires, 2**wires).')
-
-            unitary_gate = gate.DenseMatrix(wires, par[0])
-            self._circuit.add_gate(unitary_gate)
-        elif operation == 'Rot':
-            self._circuit.add_gate(gate.merge([
-                gate.RZ(wires[0], par[0]),
-                gate.RY(wires[0], par[1]),
-                gate.RZ(wires[0], par[2])
-            ]))
-        elif operation in ('CRZ', 'Toffoli', 'CSWAP'):
-            mapped_operation = self._operations_map[operation]
-            if callable(mapped_operation):
-                gate_matrix = mapped_operation(*par)
-            else:
-                gate_matrix = mapped_operation
+            self._state = QuantumState(self.num_wires)
 
-            dense_gate = gate.DenseMatrix(wires, gate_matrix)
-            self._circuit.add_gate(dense_gate)
-        else:
-            mapped_operation = self._operations_map[operation]
-            self._circuit.add_gate(mapped_operation(*wires, *par))
+        self._circuit = QuantumCircuit(self.num_wires)
 
-    @property
-    def state(self):
-        return self._state.get_vector()
+    def apply(self, operations):
+
+        for i, op in enumerate(operations):
+            # revert the wire numbering such that it adheres to qulacs
+            wires = op.wires.tolist()[::-1]
```

Comment:
```suggestion
            wires = op.wires[::-1]
```

Adapt to PennyLane master, which now passes a list to the devices (temporarily).